src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/MetricsCollectorToolSpark.java

```diff
@@ -0,0 +1,94 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines.metrics;
+
+import htsjdk.samtools.SAMFileHeader;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.metrics.MetricsArgs;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Base class for standalone Spark metrics collector tools.ß
+ *
+ */
+public abstract class MetricsCollectorToolSpark<T extends MetricsArgs>
```

Comment:
It just seems a bit strange and confusing that the class implementing this tool calss itself can be used as a MetricsCollector since it essentially just wraps one.  I don't think I would have understood how to implement one without seeing your examples.  

I'd like it if there was a way to have this operate on a provided MetricsCollector but it is a bit awkward to refactor it in a way that forces the subclasses to provide it with one.  

If you think this is the best way then keep it how it is.